python/pyarrow/_flight.pyx

```diff
@@ -986,14 +986,18 @@ cdef class MetadataRecordBatchWriter(_CRecordBatchWriter):
         ----------
         batch : RecordBatch
         """
+        cdef:
+            shared_ptr[const CKeyValueMetadata] custom_metadata
```

Comment:
Do we want to allow passing the custom_metadata as an argument to `write_batch` here?

Reply:
I am not very familiar with arrow flight API, previously I want to enhance `write_batch` here by allowing `custom_metadata` as a param, however, I find there is already an API called `write_with_metadata` (in `_flight.pyx` as well), so I am not sure if it is the right thing to do. So far this `custom_metadata` in this API is always null. Do you have any suggestion here?

Reply:
@lidavidm What do you think?

Reply:
Flight has a separate concept of metadata (I think they may have been added around the same time?) so it may make sense to expose both - I think there have been requests for that in the past. That said I think it's ok to punt to a separate PR as well

Reply:
@pitrou do you think if we can put the arrow flight related improvement in another PR? I am glad to work on it later (about arrow flight's improvement on this part) but I will probably have to learn more about arrow flight later to give it a try.

Reply:
Yes, we can.